File: demo/src/main/java/com/example/demo/Controller.java
package com.example.demo;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;


import java.util.*;

@org.springframework.stereotype.Controller
@RestController
public class Controller {

    @Autowired
    private Repository repository;



    @PostMapping("/cadastrar")
    public ResponseEntity cadastrar(@RequestBody Usuario usuario) {
        repository.save(usuario);
        return ResponseEntity.accepted().build();
    }

    @GetMapping("/cadastrados")
    public ResponseEntity cadastrados(){
        List<Usuario> lista = repository.findAll();


        return ResponseEntity.ok().body(lista);

    }


    @GetMapping("/primeiro")
    public ResponseEntity primeiro(){
        List<Usuario> lista = repository.findAll();
        Usuario usuario = lista.get(0);

        return ResponseEntity.ok().body(usuario);
    }
}

File: demo/src/main/kotlin/controllers/Controller.kt
package controllers

import models.User
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.HttpHeaders
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RestController
import repositories.Repository

@RestController
class Controller {

    @Autowired
    lateinit var repository: Repository

    @PostMapping("/cadastrar")
    fun cadastrar(@RequestBody user: User): ResponseEntity<User> {
        try {
            repository.save(user)
            return ResponseEntity.accepted().build()
        } catch (e: Exception) {
            return ResponseEntity.status(406).build()
        }
    }

}